### PR TITLE
wolfmax4k: rebuild search

### LINKED ISSUE #14511: [req]: Wolfmax4k
### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Is there already a request for your tracker?

- [X] I have checked older issues, open and closed

### Type

Public

### Tracker details

```markdown
- Name: Wolfmax4k
- URL: https://wolfmax4k.com/
- Description: Wolfmax4k is a SPANISH Public tracker for MOVIES / TV
- Minimum Seeding Time and Minimum Ratio (if Private): 
- Additional Comments: 
```

### PR DESCRIPTION
**Pull request by highfredo**
#### Description
Fix wolfmax4k search

#### Issues Fixed or Closed by this PR
* Fixes #14511

### CODE REVIEW

File: src/Jackett.Common/Indexers/Wolfmax4K.cs [hunk 1]
```diff
@@ -103,56 +120,50 @@ public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken
...
-            var maxPages = query.SearchTerm.IsNullOrWhiteSpace() ? 1 : 3;
+            var body = new Dictionary<string, string>
+            {
+                // wolfmax category&quality search is broken, do not use
+                { "pg", "" },
+                { "token", searchToken },
+                { "cidr", "" },
+                { "c", "0" },
+                { "q", query.SearchTerm },
+                { "l", query.SearchTerm.IsNullOrWhiteSpace() ? "100" : "1000" },
```
[the remark this answers is not in the document]
Comment (highfredo) [reply src/Jackett.Common/Indexers/Wolfmax4K.cs:135]: When you search with a blank query, you get the latest results, so 100 is fine.

But when searching by keyword, you can't search by episode number (neither in the format S01E01 nor Cap.101), and  the quality filter is broken. So, '100' results that then need to be filtered locally might not be enough.

Imagine a tv show that is uploaded in 480p, 720p, 1080p and 4k, if you limit the search to 100 elements you could only get 25 episodes

File: src/Jackett.Common/Indexers/Wolfmax4K.cs [hunk 2]
```diff
@@ -248,19 +241,22 @@ private ReleaseInfo ExtractReleaseInfo(IElement cardElement, TorznabQuery query)
             // https://wolfmax4k.com/descargar/serie-1080p/historial-delictivo/temporada-1/capitulo-02/
             // https://wolfmax4k.com/descargar-pelicula/avatar-v-extendida/bluray-1080p/
 
-            var quality = cardElement.QuerySelector(".quality")?.Text().Trim();
-            if (quality.IsNullOrWhiteSpace())
+            var torrentName = item.SelectToken("torrentName")?.ToString();
+            var guid = item.SelectToken("guid")?.ToString();
+            var quality = item.SelectToken("calidad")?.ToString();
```
[the remark this answers is not in the document]
Comment (highfredo) [reply src/Jackett.Common/Indexers/Wolfmax4K.cs:246]: sure!

**Comment by highfredo** [comment]
It seems they have changed how URLs are generated (or at least I hadn't noticed before) for the 'programas-tv' and 'telenovelas' categories. Honestly, I'm not sure how to parse them because, in the end, they are like TV shows but don't contain the season and episode information in the URL. They follow the format 'Cap.1251'. Is it S01E251, S12E51, or S125E01?

Ex:
https://wolfmax4k.com/descargar/programas-tv/091-alerta-policia/hdtv-720p-ac3-5-1/
https://wolfmax4k.com/descargar/telenovelas/karagul/hdtv/karagul-tierra-de-secretos/2024-06-12/

I'll check it as soon as I can, but I think it's better to leave it for another PR, so at least the rest of the searches work

**Comment by garfield69** [comment]
the caps. are likely to be the absolute episode number, much like most anime

**Comment by jordips** [comment]
Is not a fact, but most of the times you see [Cap. YXX} naming convention in Spain is Y number season and XX number of chapter of the season. Not sure what happens in a 2-digit season (I've never seen).

Seems that standard naming convention like SYYEXX is not a favourite one :(

**Comment by jordips** [comment]
Thinking about it... maybe a good solution to get chapter/season is a regex to detect cap.NNN or cap.NNNN and assume that 3-digit cap has 1 digit for season a 2 for chapter , and a 4-digit cap has 2 digits for season and 2 for chapter.

Anyway, thanks for your contribution.

**Comment by garfield69** [comment]
v0.22.119

**Comment by igebara** [comment]
Wolfmax is not working anymore. Any info?

**Comment by garfield69** [comment]
The indexer is working for me.
Raise an issue ticket if you want help.

**Pull request merged**